src/PactNet/PactNet.csproj

```diff
@@ -51,4 +51,9 @@
     <PackageReference Include="System.Runtime.Loader" Version="4.3.0" />
     <PackageReference Include="System.Threading.Thread" Version="4.3.0" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
+    <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions">
```

Comment:
Similarly to the other ASP.Net Core reference - I don't think these references should be added to the main package. They tie us to a single web framework instead of allowing the user to provide their own.

Reply:
I separated the middleware into its own project. A couple of things:
- We will need to think of a alternate way of invoking scenario for other .net framework version. (honestly, going with a message controller is not a good idea ^^)
- We could then add the provider middleware into this new project as well to be re-used.